core/domain/search_services.py

```diff
@@ -346,3 +354,137 @@ def clear_collection_search_index() -> None:
     many entries in the index.
     """
     platform_search_services.clear_index(SEARCH_INDEX_COLLECTIONS)
+
+
+class BlogPostSummaryDomainSearchDict(TypedDict):
+    """Dictionary representing the search dictionary of a blog post summary
+    domain object.
+    """
+
+    id: str
+    title: str
+    tags: List[str]
+    author_username: Optional[str]
+    rank: int
```

Comment:
Rank is used to decide the position of the particular search result in the list of search results! After statistics functionality is added (It will be done in milestone 2), rank will be calculated using the number of reads and views of the blog post. Therefore the blog post which is being read by more users will appear first in the search results as compared to others!